app/exec/extension/_lib/publish.ts

```diff
@@ -293,9 +295,7 @@ export class PackagePublisher extends GalleryBase {
 	 * @return Q.Promise that is resolved when publish is complete
 	 */
 	public publish(): Promise<GalleryInterfaces.PublishedExtension> {
-		const extPackage: GalleryInterfaces.ExtensionPackage = {
-			extensionManifest: fs.readFileSync(this.settings.vsixPath, "base64"),
-		};
+		const extPackage = fs.createReadStream(this.settings.vsixPath);
```

Comment:
OnPrem doesn't support using octet streams for the vsix package. You need to revert this (and related changes).